java/test/src/main/java/org/ray/api/test/ReferenceCountingTest.java

```diff
@@ -141,17 +141,18 @@ public void testDependencyRefCounts() {
 
   public void testBasicNestedIds() {
     RayObject<byte[]> inner = Ray.put(new byte[40 * 1024 * 1024]);
-    RayObject<List<RayObject<byte[]>>> outer = Ray.put(Collections.singletonList(inner));
+    ObjectId innerId = inner.getId();
+    checkRefCounts(ImmutableMap.of(innerId, new long[] {1, 0}));
 
-    byte[] innerObjecrIdBytes = inner.getId().getBytes();
-    inner = null;
-    System.gc();
+    RayObject<List<RayObject<byte[]>>> outer = Ray.put(Collections.singletonList(inner));
+    checkRefCounts(ImmutableMap.of(innerId, new long[] {2, 0},
+            outer.getId(), new long[] {1, 0}));
 
-    for (int i = 0; i < 100; i++) {
-      Ray.put(new byte[40 * 1024 * 1024]);
-    }
```

Comment:
The plasma store will not be fully filled and LRU eviction won't be triggered. Even if you don't create the outer object, the inner object is still available after the local reference is removed.